File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if(prices.size() == 1)
            return 0;
        int diff;
        int mini = prices[0];
        for(int i = 1; i < prices.size(); i++)
        {
            if(prices[i] < mini)
            {
                mini = prices[i];
            }
            else
            {
                diff = max(diff, prices[i] - mini);
            }
        }
        return diff;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        if(prices.length == 1)
            return 0;
        int mini = prices[0];
        int profit = 0;
        for(int i = 1; i < prices.length; i++)
        {
            if(prices[i] < mini)
                mini = prices[i];
            else
                profit = Math.max(profit, prices[i] - mini);
        }
        return profit;
    }
}